src/Components/CreateImageWizard/utilities/requestMapper.ts

```diff
@@ -526,6 +534,24 @@ const getServices = (state: RootState): Services | undefined => {
   };
 };
 
+const getUsers = (state: RootState): User[] | undefined => {
+  const users = selectUsers(state);
+  if (!users || users.length === 0) {
+    return undefined;
+  }
+  return users.map((user) => ({
+    name: user.name,
+    ...(user.password !== undefined && { password: user.password }),
+    ...(user.confirmPassword !== undefined && {
```

Comment:
I'm a bit confused here. This function, `getUsers` gets called by `getCustomizations` which in turn is called by `mapRequestFromState`. It is responsible for generating the request that is sent to the image-builder API.

It returns an array of `Users`. A `User` has the following fields:

```javascript
export type User = {
  name: string;
  /** List of groups to add the user to. The 'wheel' group should be added explicitly, as the
    default value is empty.
     */
  groups?: string[];
  ssh_key?: string;
  /** Plaintext passwords are also supported, they will be hashed and stored using the SHA-512 algorithm.
    The password is never returned in the response.
    Empty string can be used to remove the password during update but only with ssh_key set.
     */
  password?: string;
};
```

It does not have a `confirmPassword` field or an `administrator` field.

Note that to make a user an administrator, you need to add "wheel" to the user's `groups` field as shown in this image-builder API example:
https://github.com/osbuild/image-builder/blob/9bcc1043e0b661bb34dbbe9929e29068e787d029/internal/v1/api.yaml#L1929

Unfortunately I don't think Typescript can flag this because of how its type compatibility works:
https://www.typescriptlang.org/docs/handbook/type-compatibility.html

Reply:
if I remove the confirmPassword and Administrator it will not appear in the post request, when I create the blueprint 
<img width="372" alt="Screenshot 2024-12-03 at 22 30 38" src="https://github.com/user-attachments/assets/460e5707-9596-43ed-9663-e170b82c8c6b">